### PR TITLE
twister: doc: clarify harness and build_only keywords

### PR DESCRIPTION
**Pull request by nashif**
Clarify the harness keyword in test metadata.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]

File: doc/develop/test/twister.rst [hunk 1]
```diff
@@ -384,11 +384,26 @@ sysbuild: <True|False> (default False)
...
-    simple as a loopback wiring or a complete hardware test setup for
-    sensor and IO testing.
-    Usually pertains to external dependency domains but can be anything such as
-    console, sensor, net, keyboard, Bluetooth or pytest.
+    A harness string is needed to run the tests successfully. Harness is a
+    feature of twister and implemented by twister, some harnesses are defined as
+    placeholders and have no implementation yet. A harness can be seen as the
+    handler that needs to be implemented in twister to be able to evaluate test
+    pass criteria. For example, a keyboard harness is set on tests that require
+    keyboard interaction to reach verdict wheterh a test has passed or failed.
```
Comment (PerMac) [inline doc/develop/test/twister.rst:392]: typo in whether

File: doc/develop/test/twister.rst [hunk 2]
```diff
@@ -384,11 +390,26 @@ sysbuild: <True|False> (default False)
     in tests requiring sysbuild support being skipped.
 
 harness: <string>
-    A harness string needed to run the tests successfully. This can be as
-    simple as a loopback wiring or a complete hardware test setup for
-    sensor and IO testing.
-    Usually pertains to external dependency domains but can be anything such as
-    console, sensor, net, keyboard, Bluetooth or pytest.
+    A harness string is needed to run the tests successfully. Harness is a
```
Comment (pdietl) [inline doc/develop/test/twister.rst:393]:
```suggestion
    A harness string identifies a Twister harness needed to run a test successfully. A harness is a
```

File: doc/develop/test/twister.rst [hunk 3]
```diff
@@ -384,11 +390,26 @@ sysbuild: <True|False> (default False)
...
+    feature of twister and implemented by twister, some harnesses are defined as
+    placeholders and have no implementation yet. A harness can be seen as the
+    handler that needs to be implemented in twister to be able to evaluate if test
+    pass criteria. For example, a keyboard harness is set on tests that require
+    keyboard interaction to reach verdict whether a test has passed or failed.
```
Comment (pdietl) [inline doc/develop/test/twister.rst:398]:
```suggestion
    feature of Twister and is implemented by Twister, some harnesses are defined as
    placeholders and have no implementation yet. A harness can be seen as the
    handler that needs to be implemented in Twister for Twister to be able to evaluate a test's
    pass criteria. For example, a keyboard harness is set on tests that require
    keyboard interaction to reach verdict on whether a test has passed or failed. The console harness tells Twister to parse a test's text output for a regex defined in the test's YAML file.
```

**Review by pdietl — CHANGES_REQUESTED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: doc/develop/test/twister.rst [hunk 4]
```diff
@@ -467,7 +475,7 @@ harness_config: <harness configuration options>
         are i2c_hts221, i2c_bme280, i2c_FRAM, ble_fw and gpio_loop.
 
         Only one fixture can be defined per testcase and the fixture name has to
-        be unique to the tests where it is defined.
+        be unique across all tests in the test suite.
```
Comment (henrikbrixandersen) [inline doc/develop/test/twister.rst:478]: Should be included in the previous commit instead?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

File: doc/develop/test/twister.rst [hunk 5]
```diff
@@ -327,8 +327,16 @@ extra_configs: <list of extra configurations>
 
 
 build_only: <True|False> (default False)
-    If true, don't try to run the test even if the
-    selected platform supports it.
+    If true, twister will not try to run the test even if the test is buildable on the platform.
```
Comment (PerMac) [inline doc/develop/test/twister.rst:330]:
```suggestion
    If true, twister will not try to run the test even if the test is runnable on the platform.
``` 
? Sorry, just caught it at the last minute...

**Review by PerMac — CHANGES_REQUESTED** [review]

**Review by PerMac — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**